src/main/java/io/github/anthorx/parquet/sql/api/JDBCReader.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.github.anthorx.parquet.sql.api;
+
+import io.github.anthorx.parquet.sql.jdbc.model.SQLField;
+import io.github.anthorx.parquet.sql.jdbc.model.SQLRow;
+
+import javax.sql.DataSource;
+import java.sql.*;
+
+public class JDBCReader implements AutoCloseable {
+
+  private final SQLField[] columns;
+  private final int columnCount;
+  private final PreparedStatement preparedStatement;
+  private final ResultSet resultSet;
+
+  public JDBCReader(DataSource dataSource, String tableName, int fetchSize) throws SQLException {
```

Comment:
As for the writer, Connection is enough here as you don't use the Datasource object

Reply:
true as well